drivers/ethernet/eth_mcux.c

```diff
@@ -133,6 +145,25 @@ static void eth_mcux_decode_duplex_and_speed(u32_t status,
 	}
 }
 
+static struct net_if *get_iface(struct eth_context *ctx, u16_t vlan_tag)
+{
+#if defined(CONFIG_NET_VLAN)
+	struct ethernet_context *eth_ctx = net_if_l2_data(ctx->iface);
```

Comment:
it's a bad idea to expose ethernet_context in drivers, this is an L2 part. We use it in NET_DEVICE_INIT() but that should stop there. Instead, net_eth_get_vlan_iface() should take the iface pointer and the vlan_tag. 
This function get_iface() will be then simpler as well.

So you comment above about iface pointer not being useful when vlan is enabled could go away.

Reply:
I feel that we are adding too much function calls in this hot path as this function will be called for every incoming packet. Well, lets see how well this will work out, we can change this of course if this becomes a real bottleneck.

Reply:
At least what I propose is not going to do any harm: It's exactly what you did, just hiding back the ethernet_context in ethernet L2 as it should be.